src/renderer/ducks/beacon/sagas.ts

```diff
@@ -94,10 +99,82 @@ export function* pullDockerImage(
     }
 }
 
+type BeaconScanWSC = {
+    // eslint-disable-next-line camelcase
+    current_epoch: number;
+    // eslint-disable-next-line camelcase
+    is_safe: boolean;
+    // eslint-disable-next-line camelcase
+    ws_checkpoint: string;
+    // eslint-disable-next-line camelcase
+    ws_period: number;
+};
+
+function* getWeakSubjectivityCheckpoint(
+    type: WeakSubjectivityCheckpoint,
+    meta: string,
+    network: string,
+    // eslint-disable-next-line camelcase
+): Generator<CallEffect | Promise<string>, string, BeaconScanWSC & string & {current_finalized_epoch: number}> {
+    if (type !== WeakSubjectivityCheckpoint.none) {
+        if (type === WeakSubjectivityCheckpoint.custom) return meta;
+        if (type === WeakSubjectivityCheckpoint.infura) {
+            const config = getNetworkConfig(network);
+            const api = new CgEth2ApiClient((config as unknown) as IBeaconConfig, meta);
+            return yield call(api.beacon.state.getWeakSubjectivityCheckpoint);
+        }
+        if (type === WeakSubjectivityCheckpoint.beaconScan) {
+            const httpClient = new HttpClient(`https://beaconscan.com/`);
+            if (network === "mainnet") {
+                const ws = yield httpClient.get(`ws_checkpoint`);
+                if (ws) return `${ws.ws_checkpoint}:${ws.current_epoch}`;
+            }
+
+            // TODO: change scraper with api after etherscan implement it
+            const dom = yield httpClient.get(network);
+            const home = cheerio.load(dom);
```

Comment:
lol, we could have skipped this but ok 😅 
Could you write some e2e tests for this as they can change css at any point

Reply:
with cron like on dappateer?

Reply:
nah, just regular ones.